code/modules/spells/spell_types/conjure_item/_conjure_item.dm

```diff
@@ -20,28 +24,26 @@
 
 	return ..()
 
-/datum/action/cooldown/spell/conjure_item/is_valid_target(atom/cast_on)
-	return iscarbon(cast_on)
 
 /datum/action/cooldown/spell/conjure_item/cast(mob/living/carbon/cast_on)
 	if(delete_old && LAZYLEN(item_refs))
 		QDEL_LAZYLIST(item_refs)
 
 	var/obj/item/existing_item = cast_on.get_active_held_item()
-	if(existing_item)
+	if(existing_item && drop_inhand)
 		cast_on.dropItemToGround(existing_item)
 
 	var/obj/item/created = make_item()
 	if(QDELETED(created))
 		CRASH("[type] tried to create an item, but failed. It's item type is [item_type].")
 
-	cast_on.put_in_hands(created, del_on_fail = TRUE)
+	cast_on.put_in_hands(created, del_on_fail = delete_on_failure)
 	return ..()
 
 /// Instantiates the item we're conjuring and returns it.
-/// Item is made in nullspace and moved out in cast().
+/// Item is made on the owner's turf and moved into the owner's hand if the spell and mob type allows it.
 /datum/action/cooldown/spell/conjure_item/proc/make_item()
-	var/obj/item/made_item = new item_type()
+	var/obj/item/made_item = new item_type(get_turf(owner))
```

Comment:
I believe you still need to change subtype implementations of `make_item()` to also not create in nullspace, like Lesser Summon Guns